### PR TITLE
Security management related fixes

### PR DESCRIPTION
**Pull request by dgutierr**
Fixes+tests for some of the issues reported recently by QE

@romartin Can you review and ack please?

### CODE REVIEW

**Comment by romartin** [comment]
+1

File: uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java [hunk 1]
```diff
@@ -47,10 +52,12 @@ public static WebAppSettings get() {
     public void setRootDir(String dir) {
         this.rootDir = dir != null ? dir.trim() : null;
         if (rootDir != null) {
-            if (rootDir.endsWith("/")) {
-                this.rootDir = rootDir.substring(0, dir.length() - 1);
-            }
-            if (rootDir.endsWith("\\")) {
+            // Format the directory according to the file system separator
+            rootDir = StringUtils.replace(rootDir, "\\", separator);
```
Comment (ederign) [inline uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java]: Could you please remove this comment and extract to a method?

File: uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java [hunk 2]
```diff
@@ -47,10 +52,12 @@ public static WebAppSettings get() {
...
+            rootDir = StringUtils.replace(rootDir, "/", separator);
+
+            // Remove the latest separator
```
Comment (ederign) [inline uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java]: Could you please remove this comment and extract to a method?

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java [hunk 3]
```diff
@@ -302,21 +314,22 @@ void onDeleteGroupEvent(@Observes final OnDeleteEvent onDeleteEvent) {
 
     void onHomePerspectiveChangedEvent(@Observes final HomePerspectiveChangedEvent event) {
         if (checkEventContext(event, groupEditor.getAclSettings())) {
-            setDirty(true);
+            selectedHomePerspective = event.getPerspective();
+            checkStatus();
             GWT.log("Home perspective: " + event.getPerspective().getIdentifier());
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java]: I know that logs if from last version, but could you please remove it?

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java [hunk 4]
```diff
@@ -302,21 +314,22 @@ void onDeleteGroupEvent(@Observes final OnDeleteEvent onDeleteEvent) {
...
         }
     }
 
     void onPriorityChangedEvent(@Observes final PriorityChangedEvent event) {
         if (checkEventContext(event, groupEditor.getAclSettings())) {
-            setDirty(true);
+            checkStatus();
             GWT.log("Group priority : " + event.getPriority());
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java]: same here.

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java [hunk 5]
```diff
@@ -327,14 +340,14 @@ void onPermissionChangedEvent(@Observes final PermissionChangedEvent event) {
 
     void onPermissionAddedEvent(@Observes final PermissionNodeAddedEvent event) {
         if (checkEventContext(event, groupEditor.getAclEditor())) {
-            setDirty(true);
+            checkStatus();
             GWT.log("New permission node: " + event.getChildNode().getNodeName());
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/group/workflow/GroupEditorWorkflow.java]: same for all logs.

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/role/workflow/RoleEditorWorkflow.java [hunk 6]
```diff
@@ -79,21 +97,22 @@ void onEditRoleEvent(@Observes final OnEditEvent onEditEvent) {
 
     void onHomePerspectiveChangedEvent(@Observes final HomePerspectiveChangedEvent event) {
         if (checkEventContext(event, roleEditor.getAclSettings())) {
-            setDirty(true);
+            selectedHomePerspective = event.getPerspective();
+            checkStatus();
             GWT.log("Home perspective: " + event.getPerspective().getIdentifier());
```
Comment (ederign) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/role/workflow/RoleEditorWorkflow.java]: Please remove the logs.

**Comment by ederign** [comment]
@dgutierr  could you please take a look on this minor fixes?

File: uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java (hunk 1 above)
Comment (dgutierr) [inline uberfire-backend/uberfire-backend-server/src/main/java/org/uberfire/backend/server/WebAppSettings.java]: Done

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/role/workflow/RoleEditorWorkflow.java (hunk 6 above)
Comment (dgutierr) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-widgets-security-management/src/main/java/org/uberfire/ext/security/management/client/widgets/management/editor/role/workflow/RoleEditorWorkflow.java]: Logs removed. Although tbh, I do not see what are the issues (if any) of having those logs.

**Comment by dgutierr** [comment]
@ederign Done.

**Comment by ederign** [comment]
thanks @dgutierr

**Pull request merged**